src/ctapipe/image/cleaning.py

```diff
@@ -459,6 +496,129 @@ def time_constrained_clean(
+        already above the picture_thresh. Used for `tailcuts_clean`.
+    min_number_picture_neighbors: `int`
+        A picture pixel survives cleaning only if it has at least this number
+        of picture neighbors. This has no effect in case keep_isolated_pixels is True.
+        Used for `tailcuts_clean`.
+    keep_isolated_pixels: `bool`
+        If True, pixels above the picture threshold will be included always,
+        if not they are only included if a neighbor is in the picture or
+        boundary. Used for `tailcuts_clean`.
+    time_limit: `float`
+        Time limit for the `time_delta_cleaning`. Set to None if no `time_delta_cleaning`
+        should be applied.
+    time_num_neighbors: int
+        Used for `time_delta_cleaning`.
+        A selected pixel needs at least this number of (already selected) neighbors
+        that arrived within a given time_limit to itself to survive this cleaning.
+    bright_cleaning_fraction: `float`
+        `fraction` parameter for `bright_cleaning`. Pixels below
+        fraction * (average charge in the 3 brightest pixels) will be removed from
+        the cleaned image. Set `bright_cleaning_threshold` to None if no
+        `bright_cleaning` should be applied.
+    bright_cleaning_threshold: `float`
+        `threshold` parameter for `bright_cleaning`. Minimum average charge
+        in the 3 brightest pixels to apply the cleaning. Set to None if no
+        `bright_cleaning` should be applied.
+    largest_island_only: `bool`
+        Set to true to get only largest island.
+    pedestal_factor: `float`
+        Factor for interleaved pedestal cleaning. It is multiplied by the
+        pedestal standard deviation for each pixel to calculate pixelwise picture
+        threshold parameters for `tailcuts_clean` considering the current background.
+    pedestal_std: `np.ndarray`
+        Pedestal standard deviation for each pixel. See
+        `ctapipe.containers.PedestalContainer`
+
+    Returns
+    -------
+    A boolean mask of *clean* pixels.
+
+    """
+    # Step 1
+    if pedestal_std is not None:
+        pedestal_threshold = pedestal_std * pedestal_factor
+        picture_thresh = np.clip(pedestal_threshold, picture_thresh, None)
+
+    # Step 2
+    mask = tailcuts_clean(
+        geom,
+        image,
+        picture_thresh=picture_thresh,
+        boundary_thresh=boundary_thresh,
+        min_number_picture_neighbors=min_number_picture_neighbors,
+        keep_isolated_pixels=keep_isolated_pixels,
+    )
+    # Check that at least one pixel survives tailcuts_clean
+    if np.count_nonzero(mask) == 0:
+        return mask
+
+    # Step 3
+    if time_limit is not None:
+        mask = apply_time_delta_cleaning(
+            geom,
+            mask,
+            arrival_times,
+            min_number_neighbors=time_num_neighbors,
+            time_limit=time_limit,
+        )
+
+    # Step 4
+    if bright_cleaning_threshold is not None:
+        mask &= bright_cleaning(
```

Comment:
You do an `&` here after returning all zeros in the case of "mean brightness below threshold". Shouldn't it rather not modify the mask in that case?